api/src/entities/Chemeq.py

```diff
@@ -1,29 +1,88 @@
-from dataclasses import asdict, dataclass, fields
-
+from pydantic import BaseModel, Field, validator
 from typing import List, Tuple
-import numpy as np
+import numpy.typing as npt
 
 import libhg
 
 
-@dataclass(frozen=True)
-class Chemeq:
-    number_of_components: int
-    component_ids: List[int]
-    number_of_reactions: int
-    feed_composition: List[float]
-    formula_matrix: List[List[int]]
-    stoichiometric_matrix: List[List[float]]
+class Chemeq(BaseModel):
+    """
+    Pydantic class which contains requests for computing simultaneous chemical and phase equilibria
```

Comment:
```suggestion
    Model for the  simultaneous chemical and phase equilibria calculation inputs
```
Since this text is used for the API documentation. API consumers does not need to know that it is a pydantic class. But only what the generated schema is describing

Reply:
Yeah, good suggestion.